kfs-prje/src/main/java/edu/arizona/kfs/module/prje/lookup/KeyLabelFactory.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2009 The Kuali Foundation.
+ * 
+ * Licensed under the Educational Community License, Version 1.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.opensource.org/licenses/ecl1.php
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package edu.arizona.kfs.module.prje.lookup;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.kuali.rice.core.api.util.ConcreteKeyValue;
+import org.kuali.rice.core.api.util.KeyValue;
+import org.kuali.rice.krad.keyvalues.KeyValuesBase;
+
+/**
+ * KeyLabelFactory provides a KeyValues implementation for UI and data 
```

Comment:
I agree that it is OK to put off this refactor, unless we find that this design is not working in KFS 6.0.
We can discuss more when you have completed the rest of the PRJE code.